### PR TITLE
Support LinearOperators 1.0, use julia 1.3 in docs and coverage, and update to version 0.1.8

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
I've created some tests just so the coverage is decent enough.
Noteworthy: I documented `armijo_wolfe` and I changed `nbWM` to `bW_max` in accordance to `bk_max`.

File: src/linesearch/armijo_wolfe.jl [hunk 1]
```diff
@@ -1,5 +1,36 @@
 export armijo_wolfe
 
+"""
+    t, good_grad, ht, nbk, nbW = armijo_wolfe(h, h₀, slope, g)
+
+Performs a line search among the `LineModel` ``h(t) = f(x + t d)``, where
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: among -> from `x` along the direction `d` as defined by the `Linemodel` ...

File: src/linesearch/armijo_wolfe.jl [hunk 2]
```diff
@@ -1,5 +1,36 @@
...
+`h₀ = h(0.0) = f(x)`, `slope = h'(0.0) = ∇f(x)ᵀd` and `g` is a which will be
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: * 0 instead of 0.0
* `g` is a ... what? vector?
* which -> that

File: src/linesearch/armijo_wolfe.jl [hunk 3]
```diff
@@ -1,5 +1,36 @@
...
+overwritten with the gradient at various points. On exit, if `good_grad=true`,
+`g` contains the gradient at the returned step length.
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: returned -> final

File: src/linesearch/armijo_wolfe.jl [hunk 4]
```diff
@@ -1,5 +1,36 @@
...
+The steplength is chosen to satisfy the Armijo and Wolfe conditions. The Armijo condition is
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: In fact the steplength only attempts to satisfy the Wolfe condition. The final steplength may only satisfy Armijo's condition.

File: src/linesearch/armijo_wolfe.jl [hunk 5]
```diff
@@ -1,5 +1,36 @@
...
+```math
+h(t) ≤ h₀ + τ₀ t slope
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: slope -> h'(0)

File: src/linesearch/armijo_wolfe.jl [hunk 6]
```diff
@@ -1,5 +1,36 @@
...
+```
+and the Wolfe condition is
+```math
+h'(t) ≤ τ₁ slope.
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: slope -> h'(0)

File: src/linesearch/armijo_wolfe.jl [hunk 7]
```diff
@@ -1,5 +1,36 @@
...
+The output is the following:
+- t: the step length;
+- good_grad: whether `g` is the gradient at `t`;
+- ht: the model at `t`, i.e., `f(x + t * d)`;
+- nbk: the number of times the steplength was decreased to satisfy the Armijo condition, i.e., number of backtracks;
+- nbW: the number of times the steplength was increased to satisfy the Wolfe condition.
+
+The following keyword arguments can be provided:
+- `t`: starting steplength (default `1`);
+- `τ₀`: acceptance argument in the Armijo condition (default `max(1e-4, √ϵₘ)`);
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: acceptance argument -> slope factor

File: src/linesearch/armijo_wolfe.jl [hunk 8]
```diff
@@ -1,5 +1,36 @@
...
+h(t) ≤ h₀ + τ₀ t slope
+```
+and the Wolfe condition is
+```math
+h'(t) ≤ τ₁ slope.
+```
+
+The output is the following:
+- t: the step length;
+- good_grad: whether `g` is the gradient at `t`;
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: t -> x + t d

File: src/linesearch/armijo_wolfe.jl [hunk 9]
```diff
@@ -1,5 +1,36 @@
...
+- ht: the model at `t`, i.e., `f(x + t * d)`;
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: model -> model value

File: src/linesearch/armijo_wolfe.jl [hunk 10]
```diff
@@ -1,5 +1,36 @@
...
+- nbk: the number of times the steplength was decreased to satisfy the Armijo condition, i.e., number of backtracks;
+- nbW: the number of times the steplength was increased to satisfy the Wolfe condition.
+
+The following keyword arguments can be provided:
+- `t`: starting steplength (default `1`);
+- `τ₀`: acceptance argument in the Armijo condition (default `max(1e-4, √ϵₘ)`);
+- `τ₁`: acceptance argument in the Wolfe condition (default `0.9999`):
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: Need τ₁ > τ₀

File: test/test_linesearch.jl [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    nlp = ADNLPModel(x -> x[1]^2 + 4 * x[2]^2, ones(2))
+    x = nlp.meta.x0
+    d = -ones(2)
+    lm = LineModel(nlp, x, d)
+    g = zeros(2)
+
+    @test obj(lm, 0.0) == obj(nlp, x)
+    @test grad(lm, 0.0) == dot(grad(nlp, x), d)
+    @test grad!(lm, 0.0, g) == dot(grad(nlp, x), d)
+    @test g == grad(nlp, nlp.meta.x0)
```
Comment (dpo) [inline test/test_linesearch.jl]: x vs nlp.meta.x0

File: src/linesearch/armijo_wolfe.jl [hunk 12]
```diff
@@ -1,5 +1,40 @@
 export armijo_wolfe
 
+"""
+    t, good_grad, ht, nbk, nbW = armijo_wolfe(h, h₀, slope, g)
+
+Performs a line search from `x` along the direction `d` as defined by the
+`LineModel` ``h(t) = f(x + t d)``, where
```
Comment (dpo) [inline src/linesearch/armijo_wolfe.jl]: What do double backticks do?

Comment (abelsiqueira) [reply src/linesearch/armijo_wolfe.jl]: Math-like environment. Only evident in the docs.

Comment (dpo) [reply src/linesearch/armijo_wolfe.jl]: Ah ok, thank you.

**Pull request merged**